libs/core/src/lib/calendar/calendar.component.ts

```diff
@@ -212,12 +212,36 @@ export class CalendarComponent<D> implements OnInit, ControlValueAccessor, Valid
      */
     _currentlyDisplayed: CalendarCurrent;
 
+    /**
+     * @hidden
+     */
+    get currentlyDisplayed(): CalendarCurrent {
+        return this._currentlyDisplayed;
+    }
+
+    set currentlyDisplayed(currentlyDisplayed: CalendarCurrent) {
+        this._currentlyDisplayed = currentlyDisplayed;
+        if (this.previousButtonDisableFunction && this.nextButtonDisableFunction) {
```

Comment:
remove the check condition from this setter and apply it inside `_setNavigationButtonsStates` function. Additionally, you should check for presence of `previousButtonDisableFunction` and `nextButtonDisableFunction` functions apart